site/examples/check-lists.js

```diff
@@ -32,7 +32,45 @@ const CheckListsExample = () => {
 }
 
 const withChecklists = editor => {
-  const { deleteBackward } = editor
+  const { deleteBackward, insertBreak } = editor
+  editor.insertBreak = () => {
+    const match = Editor.above(editor, {
+      match: n => {
+        return Editor.isBlock(editor, n)
+      },
+    })
+    if (match) {
+      const [matchingNode] = match
+      if (matchingNode.type !== 'paragraph') {
+        if (
+          matchingNode.children[matchingNode.children.length - 1].text
+            .length === 0
+        ) {
+          Transforms.delete(editor)
+          const paragraph = {
+            type: 'paragraph',
+            children: [{ text: '' }],
+          }
+          Transforms.insertNodes(editor, paragraph)
```

Comment:
What about:

`Transforms.setNodes(editor, { type: 'paragraph});`

Reply:
@zbeyens  As you can see below in the screenshot of second console Array position 7th if we use `Transforms.setNodes(editor, { type: 'paragraph' });`, then when an empty checklist is converted to paragraph it also contains the checklist additional property `checked:false` which is not correct in my view. 


![Screenshot 2020-01-05 at 3 13 22 AM](https://user-images.githubusercontent.com/13781801/71772026-ecd9f500-2f69-11ea-9417-dbf72e48f3ba.png)

Reply:
True, it would be nice to have an option to "reset" the node state on `setNodes` (remove additional properties) @ianstormtaylor

Reply:
@zbeyens wouldn't it be better if the `setNodes` by default removes the additional properties and just copies the `type` and empty `children` ? or is there a scenario where doing this implicitly will break the code?